resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -5,1171 +5,1691 @@
+        <xs:all>
+            <xs:element name="discovery_config" type="discoverySettingsType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="avoid_builtin_multicast" type="boolean" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="use_WriterLivelinessProtocol" type="boolean" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="metatraffic_external_unicast_locators" type="externalLocatorListType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="metatrafficUnicastLocatorList" type="locatorListType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="metatrafficMulticastLocatorList" type="locatorListType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="initialPeersList" type="locatorListType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="readerHistoryMemoryPolicy" type="historyMemoryPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="writerHistoryMemoryPolicy" type="historyMemoryPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="readerPayloadSize" type="uint32" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="writerPayloadSize" type="uint32" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="mutation_tries" type="uint32" minOccurs="0" maxOccurs="1"/>
+        </xs:all>
     </xs:complexType>
 
-    <xs:complexType name="DiscoveryServerList">
-        <xs:sequence>
-            <xs:element name="RemoteServer" type="RemoteServerAttributes" minOccurs="1" maxOccurs="unbounded"/>
+    <!--builtin Discovery Configuration:
+        ├ discoveryProtocol         [string] ("SIMPLE", "CLIENT", "SERVER", "BACKUP", "NONE"),
+        ├ discoveryServersList      [0~1],
+        |   └ RemoteServer          [1~*],
+        |      |   ╚ att. prefix    [string] REQ,
+        |      ├ metatrafficUnicastLocatorList   [0~1],
+        |      └ metatrafficMulticastLocatorList [0~1],
+        ├ ignoreParticipantFlags    [string] ("NO_FILTER", "FILTER_DIFFERENT_HOST", "FILTER_DIFFERENT_PROCESS", "FILTER_SAME_PROCESS", "FILTER_DIFFERENT_PROCESS | FILTER_SAME_PROCESS"),
+        ├ EDP                       [string] ("SIMPLE", "STATIC"),
+        ├ simpleEDP                 [0~1],
+        |   ├ PUBWRITER_SUBREADER   [bool],
+        |   └ PUBREADER_SUBWRITER   [bool],
+        ├ leaseDuration             [durationType],
+        ├ leaseAnnouncement         [durationType],
+        ├ initialAnnouncements      [0~1],
+        |   ├ count                 [uint32],
+        |   └ period                [durationType],
+        ├ clientAnnouncementPeriod  [durationType],
+        └ static_edp_xml_config     [0~*] [string] -->
+    <!-- TODO:  How to ensure that simpleEDP is defined only when EDP is set as "SIMPLE"?
+                How to ensure that static_edp_xml_config is defined only when EDP is set as "STATIC"? -->
+    <xs:complexType name="discoverySettingsType">
+        <xs:sequence minOccurs="0">
```

Comment:
Seems to happen because of the order of the elements, which should be any, but sequence tag imposes an specific order, am I right?
Happens also for the unrecognized name of some element defined in the xml file